JSON Serialization Spec (#289)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

Other specifications or profiles of this specification may define alternative formats for transporting the Combined Format for Presentation that envelopes multiple such objects into one object, and provides Key Binding using means other than the Key Binding JWT.

# JWS JSON Serialization {#json_serialization}

> (Work in progress...)
> 

Author: ```suggestion
```

Author: I added this just to show that even the sentences I wrote were just placeholders. I think we can remove this for now.

Reviewer: I left it in for similar reasons but can remove at this point.

This section describes an optional alternate format for SD-JWT using the JWS JSON Serialization from [@!RFC7515].

For both the General and Flattened JSON Serialization, the SD-JWT is represented as a JSON object according

> to Section 7.2 of [@!RFC7515]. The disclosures (both for issuance and presentation) are included in the

Reviewer 2: ```suggestion
to Section 7.2 of [@!RFC7515]. The disclosures (both for issuance and presentation) MUST be included in the
```

Reviewer: I was intentionally avoiding RFC 2119 key words like MUST here because it seemed unnecessary and potentially limiting.

Author: Do you have any specific use cases in mind where this might be limiting?

Reviewer: A profile or something might want to base64url-encode the JWS JSON part and add the disclosures `~` separated to the end.  I don't like that. And defintily don't think this spec should describe it. But didn't want to preclude it.

Author: SHOULD this be a SHOULD then?

Reviewer 2: let's keep as-is and see if the topic arises again.

serialized JWS using the key `disclosures` at the top-level of the JSON object (the same level as the `payload` member). The
value of the `disclosures` member is array of strings where each element is an individual Disclosure

> as described in (#creating_disclosures). The Issuer includes a Disclosure for each selectively
> disclosable claim of the SD-JWT payload. Whereas the Holder includes only the Disclosures
> selected for the given presentation. Additionally, for presentation with a Key Binding, the Holder adds

Author: ```suggestion
as described in (#creating_disclosures). The Issuer includes a Disclosure for each selectively
disclosable claim of the SD-JWT payload, whereas the Holder includes only the Disclosures
selected for the given presentation. Additionally, for presentation with a Key Binding, the Holder adds
```

Other specifications or profiles of this specification may define alternative formats for transporting the Combined Format for Presentation that envelopes multiple such objects into one object, and provides Key Binding using means other than the Key Binding JWT.

# JWS JSON Serialization {#json_serialization}

> This section describes an optional alternate format for SD-JWT using the JWS JSON Serialization from [@!RFC7515].

Reviewer 2: see below
